### PR TITLE
fix: Add notice for missing stop name for specific location types

### LINKED ISSUE #1078: Add a rule for missing stop name
**What problem in GTFS datasets does this new rule address? Please describe.**
The GTFS specification [mentions in `stops.stop_name`](https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#stopstxt):

> Conditionally Required:
> - Required for locations which are stops (location_type=0), stations (location_type=1) or entrances/exits (location_type=2).
> - Optional for locations which are generic nodes (location_type=3) or boarding areas (location_type=4).

**Describe the new validation rule**
We should add a rule that would trigger a `MissingStopNameNotice` if:
- (`location_type` has values 0, 1 or 2) AND (`stop_name` is missing) for a line in `stops.txt`

**Error vs warning**
ERROR

### PR DESCRIPTION
**Pull request by briandonahue**
**Summary:**

Adds validation for required stop name for STOP, STATION, ENTRANCE location types. Closes #1078 

**Expected behavior:** 

Should generate an Error notice if stop name is missing for the above location types.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s) (N/A)

### CODE REVIEW

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopNameValidator.java [hunk 1]
```diff
@@ -51,6 +63,30 @@ private boolean isValidStopDesc(String stopDesc, String stopName) {
...
+   * (http://gtfs.org/reference/static#stopstxt)
+   *
+   * <p>Severity: {@code SeverityLevel.Error}
+   */
+  static class MissingStopNameNotice extends ValidationNotice {
+
+    private final long csvRowNumber;
+    private GtfsLocationType locationType;
+    private final String stopId;
+    private final String routeDesc;
```
Comment (bdferris-v2) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopNameValidator.java:79]: routeDesc => stopDesc

Comment (bdferris-v2) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/StopNameValidator.java:79]: I'm not sure it's worth including stopDesc here.  It's not really relevant to the name.

**Review by bdferris-v2 — APPROVED** [review]
Modulo the one comment, this LGTM.

**Review by isabelle-dr — CHANGES_REQUESTED** [review]
The new rule LGTM, we just need the RULES.md update :)

More a general thought than something specific to this PR: I wonder if we want to formalize our preferred approach between:
- generating a custom notice for the conditionally required fields
- re-using `MissingRequiredField`

**Comment by briandonahue** [comment]
> The new rule LGTM, we just need the RULES.md update :)

@isabelle-dr updated!

**Comment by bdferris-v2** [comment]
@isabelle-dr apologies I missed the RULES.md update on this.  I've taken a stab at adding a unit-test to catch this sort of thing in #1331.

File: RULES.md [hunk 2]
```diff
@@ -65,6 +65,7 @@ Each Notice is associated with a severity: `INFO`, `WARNING`, `ERROR`.
 | [`missing_required_column`](#missing_required_column)                                                                             | A required column is missing in the input file.                                                                                                        |
 | [`missing_required_field`](#missing_required_field)                                                                               | A required field is missing.                                                                                                                           |
 | [`missing_required_file`](#missing_required_file)                                                                                 | A required file is missing.                                                                                                                            |
+| [`missing_stop_name`](#missing_stop_name)                                                                                         | Missing required `stop_name` for stop with location type of `0`, `1`, or `2`.                                                                                                  |
```
Comment (isabelle-dr) [inline RULES.md:68]:
```suggestion
| [`missing_stop_name`](#missing_stop_name) | `stops.stop_name` is required for `location_type` equal to `0`, `1`, or `2`. |
```

File: RULES.md [hunk 3]
```diff
@@ -1030,6 +1031,28 @@ A required file is missing. If this notice is triggered for every core file, it
 
 </details>
 
+<a name="MissingStopName"/>
+
+### missing_stop_name
+
+Missing required `stop_name` for stop with location type of `0`, `1`, or `2`.
```
Comment (isabelle-dr) [inline RULES.md:1038]:
```suggestion
`stops.stop_name` is required for locations that are stops (`location_type=0`), stations (`location_type=1`) or entrances/exits (`location_type=2`).
```

**Review by isabelle-dr — COMMENTED** [review]
Just two propositions for the definitions, thanks for updating this so quickly @briandonahue!
Why are there also modifications to the java files in this PR?

**Comment by briandonahue** [comment]
> Why are there also modifications to the java files in this PR?

@isabelle-dr this PR contains the updated validation and the RULES.md update to go along with it.  Should they be separate?

**Review by isabelle-dr — APPROVED** [review]
Sorry, I got mixed up with #1330 🙈. 
This is perfect! LGTM

**Comment by isabelle-dr** [comment]
Can you update this branch with the latest master?

**Pull request merged**